library/std/src/io/buffered/bufwriter.rs

```diff
@@ -59,9 +59,9 @@ use crate::io::{
 /// together by the buffer and will all be written out in one system call when
 /// the `stream` is flushed.
 ///
-/// [`TcpStream::write`]: Write::write
+/// [`TcpStream::write`]: crate::net::TcpStream::write
 /// [`TcpStream`]: crate::net::TcpStream
-/// [`flush`]: Write::flush
+/// [`flush`]: BufWriter::flush
```

Comment:
Weird, how come these are replaced with a different type?

Reply:
This is intentional; before it would take you to a different page instead of linking to current page. That was a regression introduced when these links were originally switched which couldn't be fixed at the time.